docs/site/Model.md

```diff
@@ -25,63 +25,92 @@ or gRPC message definitions, and vice versa.
 
 There are two subtly different types of models for domain objects:
 
+- Entity: A domain object that has an identity (ID). Its equality is based on
+  the identity. For example, `Customer` can be modeled as an `Entity` because
+  each customer has a unique customer id. Two instances of `Customer` with the
+  same customer id are equal since they refer to the same customer. For example:
```

Comment:
Nitpick: I assumed the code below would be showing how two entities are equal because they have the same `id` instead of seeing what an Entity class looks like based on the sentence before. Maybe "For example, this is how a `Customer` can be modelled:" or something

Reply:
Good suggestion 👍 I applied it to both `Entity` and `Model`.